src/services/user/services/MailRegistrationLink.js

```diff
@@ -27,53 +27,75 @@ const getCurrentUserInfo = (id) => userModel.findById(id)
 	.lean()
 	.exec();
 
+
+const getUserInfos = (ids, app) => app.service('/userModel').find({
+	query: {
+		_id: ids,
+		select: ['schoolId', 'email', 'firstName', 'lastName', 'preferences'],
+		$populate: 'roles',
+	},
+});
+
 class SendRegistrationLinkService {
 	async create(data, params) {
 		let totalMailsSend = 0;
 
 		try {
-			const { userIds } = data;
+			const { userIds: allUserIds } = data;
 			let accounts = await this.app.service('/accounts').find({
 				query: {
-					userId: userIds,
+					userId: allUserIds,
 				},
 			});
 			accounts = accounts.map((account) => String(account.userId));
+			const userIds = allUserIds.filter(
+				(id) => undefined !== accounts.find((a) => a.userId.toString() === id.toString()),
+			);
+
+			const users = await getUserInfos(userIds, this.app);
+
+			await Promise.all(users.map(async (user) => {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Expected to return a value at the end of async arrow function.](https://app.codacy.com/manual/schul-cloud/schulcloud-server/pullRequest?prid=5424589)